### PR TITLE
Add JF_RELEASES_REPO to GitHub actions

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---
* Downloading the Frogbot binary with Artifactory and remote repository when using the GitHub actions CI.

### CODE REVIEW

File: action/test/main.spec.ts [hunk 1]
```diff
@@ -25,11 +25,68 @@ describe('Frogbot Action Tests', () => {
...
+        test.each(cases)('Remote CLI Url for %s-%s', (platform, arch, fileName, expectedUrl) => {
+            myOs.platform.mockImplementation(() => <NodeJS.Platform>platform);
+            myOs.arch.mockImplementation(() => arch);
+            let cliUrl: string = Utils.getCliUrl('2', '2.8.7', fileName, releasesRepo);
+            expect(cliUrl).toBe(expectedUrl);
+        });
+    });
+
+    describe('generateAuthString', () => {
+        it('should return an empty string if releasesRepo is falsy', () => {
```
Comment (yahavi) [inline action/test/main.spec.ts:63]: Let's be consistent with the names of the other tests by capitalizing the test names.
For example:
```ts
describe('Generate auth string', () => {
```

```ts
it('Should return an empty string if releasesRepo is falsy', () => {
```

File: action/test/main.spec.ts [hunk 2]
```diff
@@ -25,11 +25,68 @@ describe('Frogbot Action Tests', () => {
...
-            let cliUrl: string = Utils.getCliUrl('1', '1.2.3', fileName);
+            let cliUrl: string = Utils.getCliUrl('1', '1.2.3', fileName, '');
             expect(cliUrl).toBe(expectedUrl);
         });
     });
 
+    describe('Frogbot URL Tests With Remote Artifactory', () => {
+        const myOs: jest.Mocked<typeof os> = os as any;
+        const releasesRepo: string = 'frogbot-remote';
+        process.env['JF_URL'] = 'https://myfrogbot.com/';
```
Comment (yahavi) [inline action/test/main.spec.ts:36]: Please consider using afterAll which cleans up all environment variables after each test

File: action/src/utils.ts [hunk 3]
```diff
@@ -96,8 +112,17 @@ export class Utils {
         core.addPath(cliDir);
     }
 
-    public static getCliUrl(major: string, version: string, fileName: string): string {
+    public static getCliUrl(major: string, version: string, fileName: string, releasesRepo: string): string {
         let architecture: string = 'frogbot-' + Utils.getArchitecture();
+        if (releasesRepo) {
+            let platformUrl: string = process.env.JF_URL ?? '';
+            if (!platformUrl) {
+                throw new Error('failed while downloading Frogbot from Artifactory, JF_URL must be set');
```
Comment (yahavi) [inline action/src/utils.ts:120]: Unlike Go, errors in Typescript are capitalized
```suggestion
                throw new Error('Failed while downloading Frogbot from Artifactory, JF_URL environment variable must be set');
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: action/test/main.spec.ts (hunk 1 above)
Comment (omerzi) [reply action/test/main.spec.ts:63]: Done

File: action/test/main.spec.ts (hunk 2 above)
Comment (omerzi) [reply action/test/main.spec.ts:36]: Done

File: action/src/utils.ts (hunk 3 above)
Comment (omerzi) [reply action/src/utils.ts]: Done

**Review by yahavi — APPROVED** [review]

**Pull request merged**